src/pip/_internal/req/req_install.py

```diff
@@ -590,29 +620,10 @@ def prepare_metadata(self):
         with indent_log():
             self.metadata_directory = metadata_generator(self)
 
-        if not self.req:
-            if isinstance(parse_version(self.metadata["Version"]), Version):
-                op = "=="
-            else:
-                op = "==="
-            self.req = Requirement(
-                "".join([
-                    self.metadata["Name"],
-                    op,
-                    self.metadata["Version"],
-                ])
-            )
+        if not self.name:
```

Comment:
This conditional is equivalent and reads more like "unnamed" requirement, which is what we care about here -- the fact that `self.req is None` for unnamed requirements is coded in `InstallRequirement.name`.